Add Wordpress example, resolves #57

LeendersR opened the pull request:
Adds the Wordpress example to resolve #57. Also includes a new version of replicatedservice.py that accepts three new properties:
- cluster_ip: to set the ip of a service
- privileged: to create privileged containers
- volumes: to mount volumes

LeendersR committed:
Adds Wordpress example

LeendersR committed:
Adds Wordpress example

Updates Wordpress example

Change hardcoded type urls from leendersr/deployment-manager to kubernetes/deployment-manager

Fix typo, we want to deploy wordpress.yaml not .jinja

Fix spelling

Updates Makefile to use generic Docker registry

Adds picture to README

Improves readability

Updates Wordpress README for Kubernetes 1.1 release

Adds extra primitive types for the Wordpress example

Move types/* to templates/*

Updates README to link to master/templates

Changes the Wordpress template so that it uses PersistentVolumeClaims

Creates an NFS type

LeendersR committed:
Adds Wordpress example

LeendersR committed:
Adds Wordpress example

Adds wordpress.yaml code to README

Changes absolute url from leendersr/deployment-manager back to kubernetes/deployment-manager

LeendersR committed:
Updates Wordpress example

LeendersR committed:
Updates Wordpress example

LeendersR committed:
Change hardcoded type urls from leendersr/deployment-manager to kubernetes/deployment-manager

LeendersR committed:
Change hardcoded type urls from leendersr/deployment-manager to kubernetes/deployment-manager

LeendersR committed:
Fix typo, we want to deploy wordpress.yaml not .jinja

LeendersR committed:
Fix typo, we want to deploy wordpress.yaml not .jinja

LeendersR committed:
Fix spelling

LeendersR committed:
Fix spelling

LeendersR commented:
I signed it!

LeendersR committed:
Updates Makefile to use generic Docker registry

LeendersR committed:
Updates Makefile to use generic Docker registry

LeendersR committed:
Adds picture to README

LeendersR committed:
Adds picture to README

bmelville commented on examples/wordpress/wordpress-resources.yaml:
```diff
@@ -0,0 +1,11 @@
+resources:
```
Please add a note here that this is a Google Cloud Deployment Manager configuration

bmelville commented on types/replicatedservice/v2/replicatedservice.py:
```diff
@@ -0,0 +1,194 @@
+"""Defines a ReplicatedService type by creating both a Service and an RC.
```
It's hard for me to tell here what's changed that's causing this version bump. Should we have some kind of notes in type versions to understand these kinds of changes?

LeendersR replied:
Maybe add a changelog field in the schema?

bmelville replied:
It may make sense to have a separate file that is part of the "type" definition, since the changelog isn't really necessary to the schema of the template.

Would be curious to see what @jackgr thinks about this, as he's given the semver stuff a lot of thought and is a strong proponent.

bmelville commented on examples/wordpress/wordpress.jinja:
```diff
@@ -0,0 +1,81 @@
+{% set MYSQL_PASSWORD = MYSQL['password'] or 'mysql-password' %}
+{% set MYSQL_DISK = MYSQL['disk'] or 'mysql-disk' %}
+{% set MYSQL_DISK_FSTYPE = MYSQL['fstype'] or 'ext4' %}
+imports:
```
newlines between variables and imports, as well as imports and resources, just for readability

LeendersR committed:
Improves readability

LeendersR committed:
Improves readability

zmerlynn commented on examples/wordpress/README.md:
````diff
@@ -0,0 +1,160 @@
+```
+
+### Privileged containers
+To use NFS we need to be able to launch privileged containers. If your Kubernetes cluster doesn't support this you can either manually change this in every Kubernetes minion or you could set the flag at `kubernetes/saltbase/pillar/privilege.sls` to true and (re)launch your Kubernetes cluster. Once Kubernetes 1.1 releases this should be enabled by default.
````
1.1 has launched, you can remove this section.

zmerlynn commented on examples/wordpress/README.md:
````diff
@@ -0,0 +1,160 @@
+### NFS Library
+Currently the nfs-common library should be installed by default on Kubernetes nodes. In case nfs-common is not installed you can install it on all Kubernetes nodes using the following command:
+```
+gcloud compute instances list | cut -d ' ' -f 1 | tail -n +2 | xargs -n1 gcloud compute ssh --command="sudo apt-get update;sudo apt-get -y install nfs-common"
````
If you're going to give `gcloud` commands, then `nfs-common` is installed (now), unless they happen to be bringing up CoreOS nodes. I ensured it before 1.1 shipped.

zmerlynn commented on examples/wordpress/README.md:
```diff
@@ -0,0 +1,160 @@
+    volumes:
+      - mount_path: /var/www/html
+        nfs:
+          server: {{ NFS_SERVER_IP }}
```
Rather than duplicate the `NFS_SERVER_IP` twice, create a single `PersistentVolume` object for the NFS server and just reuse a named `ReadWriteMany` `PersistentVolumeClaim` object. This makes the `NFS_SERVER_IP` templated in exactly one place and reduces the grossness (see the example I dropped in https://github.com/kubernetes/kubernetes/tree/master/examples/nfs ).

LeendersR committed:
Updates Wordpress README for Kubernetes 1.1 release

LeendersR committed:
Updates Wordpress README for Kubernetes 1.1 release

LeendersR committed:
Adds extra primitive types for the Wordpress example

LeendersR committed:
Adds extra primitive types for the Wordpress example

LeendersR committed:
Merge remote-tracking branch 'upstream/master'

LeendersR committed:
Move types/* to templates/*"

LeendersR committed:
Move types/* to templates/*"

zmerlynn commented:
@LeendersR: BTW, it's anti-recommended to push up a merge commit. You should instead always rebasing from `upstream/master` (basically never, ever use `git pull`, it's a waste of a command - unless you reconfigure it to default to rebase). Right now, if this PR is merged, you're effectively merging back a merge from HEAD, which forms an odd little Gordian knot in history and is just confusing to readers, so please squash to the effective delta and repush. (Be careful.)

LeendersR committed:
Updates README to link to master/templates

LeendersR committed:
Updates README to link to master/templates

LeendersR committed:
Changes the Wordpress template so that it uses PersistentVolumeClaims

LeendersR committed:
Changes the Wordpress template so that it uses PersistentVolumeClaims

LeendersR committed:
Creates an NFS type

LeendersR committed:
Creates an NFS type

LeendersR committed:
Merge branch 'master' of github.com:LeendersR/deployment-manager

LeendersR committed:
Merge branch 'master' of github.com:LeendersR/deployment-manager

LeendersR committed:
Updates README and adds clain-name property to the NFS type

jiangyaoguo committed:
fix namespace in install.yaml

LeendersR committed:
Adds wordpress.yaml code to README

LeendersR committed:
Changes absolute url from leendersr/deployment-manager back to kubernetes/deployment-manager

Commit:
Refactoring CreateDeployment to isolate manifest creation logic.

CreateDeployment will now create a deployment before expansion, and will
fail the deployment on expansion error.

LeendersR commented:
@zmerlynn Thanks for the tip.
@bmelville Any comments?

Commit:
Adding Expand to the manager.

Commit:
PutDeployment also uses createManifest and sets errors appropriately.

Commit:
Adding Expand endpoint as "/expand".

This endpoint takes a template and returns a manifest containing only
the expanded config and layout.

Commit:
Adding the rest of validations for manager expand test.

Commit:
Merge pull request #91 from bmelville/39

Implements correct Expand in API and client

Commit:
Merge pull request #86 from jiangyaoguo/fix-namespace-in-install-file

fix namespace in install.yaml

Commit:
Merge pull request #88 from bmelville/23

Supports describing a type

LeendersR committed:
Adds Wordpress example
